File: ios-client/GSMRelayClientIOS/Models/ServerConfig.swift
import Foundation

struct ServerConfig: Codable, Equatable {
    var baseURL: String
    var username: String
    var password: String
    var gatewayId: String
    var masterPhone: String

    static let defaults = ServerConfig(
        baseURL: "http://86.120.150.58:5174",
        username: "admin",
        password: "admin1316",
        gatewayId: "pQF6bci9",
        masterPhone: "0724264464"
    )

    var isValid: Bool {
        !baseURL.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty &&
        !username.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty &&
        !password.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }

    var normalizedBaseURL: String {
        let trimmed = baseURL.trimmingCharacters(in: .whitespacesAndNewlines)
        if trimmed.hasSuffix("/") {
            return String(trimmed.dropLast())
        }
        return trimmed
    }
}

File: android/app/src/main/java/com/security/gsmrelay/data/model/ServerConfig.kt
package com.security.gsmrelay.data.model

data class ServerConfig(
    val baseUrl: String = "",
    val username: String = "",
    val password: String = "",
    val gatewayId: String = "",
    val masterPhone: String = ""
) {
    fun isValid(): Boolean {
        return baseUrl.isNotBlank() && username.isNotBlank() && password.isNotBlank()
    }

    companion object {
        const val DEFAULT_BASE_URL = "http://86.120.150.58:5174"
        const val DEFAULT_USERNAME = "admin"
        const val DEFAULT_PASSWORD = "admin1316"
        const val DEFAULT_GATEWAY_ID = "pQF6bci9"
        const val DEFAULT_MASTER_PHONE = "0724264464"
    }
}
